Remove control flow instruction filter when `use_fractional_gates=true`

kt474 opened the pull request:
<!--
⚠️ The pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

Now that backends support both fractional gates and dynamic circuits we don't need to filter out the control flow instructions when `use_fractional_gates=true` is used. 

### Details and comments
Fixes #2363

jyu00 commented:
> For `get_translation_stage_plugin()` I think we'll have to deprecate it first even though it's not used anywhere
> 
> https://quantum.cloud.ibm.com/docs/en/api/qiskit-ibm-runtime/ibm-backend#get_translation_stage_plugin

Isn't it automatically picked up when transpiling using an IBMBackend ? The `IBMFractionalTranslationPlugin` says 

https://github.com/Qiskit/qiskit-ibm-runtime/blob/d44ecbde9845160b4706b8a1a9ce2d40716fa356/qiskit_ibm_runtime/transpiler/plugin.py#L120-L121

which sounds to me trying to transpile with both fractional gates & control flow may blow up.

kt474 commented:
> https://github.com/Qiskit/qiskit-ibm-runtime/blob/3d78e094c7c597f4050b797d5ea8c6152741e6cc/qiskit_ibm_runtime/ibm_backend.py#L496-L500

@nkanazawa1989 How should this be handled now that our backends support jobs with both fractional gates are conditionals?

kt474 commented:
> This PR looks good to me. Just minor comments.

Thanks! Addressed comments and deprecated `IBMDynamicTranslationPlugin` and `IBMFractionalTranslationPlugin`

ElePT commented on qiskit_ibm_runtime/ibm_backend.py:
```diff
@@ -495,9 +495,7 @@ def run(self, *args, **kwargs) -> None:  # type: ignore[no-untyped-def]
-        if not self.options.use_fractional_gates:
-            return "ibm_dynamic_circuits"
-        return "ibm_fractional"
+        return "ibm_dynamic_and_fractional"
```
If fractional gates can still be disabled (correct me if I'm wrong), I think that it still make senses to keep the `ibm_dynamic_circuits` standalone plugin, right? So having something like:

```suggestion
        if not self.options.use_fractional_gates:
                return "ibm_dynamic_circuits"
        return "ibm_dynamic_and_fractional"
```

In this case I would "undeprecate" the ibm_dynamic_circuits.

ElePT reviewed (commented):
Thanks Kevin! I have one final comment after taking a look at the whole PR again. I don't know why I assumed that `use_fractional_gates` would no longer be used, but now I realized that we still have 2 cases for the transpiler plugin.
